### ISSUE TITLE
Closed plotter with a GIF writer makes subsequent plotter hang

### ISSUE THREAD

**Issue by adeak** [pyvista/pyvista#3139]
### Describe the bug, what's wrong, and what you expected.

If I take a plotter that writes a GIF, then `.close()` it, then open another plotter and `.show()` that, the call to `.show()` never returns even if I close the second figure window. This hangs the python script.

### Steps to reproduce the bug.

Example code that hangs on my system on `upstream/main` (but also `v0.36.1`):
```py
import numpy as np
import pyvista as pv

x, y = np.mgrid[:5.0, :5.0]
z = np.zeros_like(x)

plotter = pv.Plotter()
plotter.open_gif('tmp.gif')
plotter.add_mesh(pv.StructuredGrid(x, y, z))
plotter.write_frame()
plotter.close()

plotter = pv.Plotter()
plotter.add_mesh(pv.StructuredGrid(x, y, z))
plotter.show()
```

Adding temporary prints reveals that the second plotter is somehow stuck in the interactive event loop:
https://github.com/pyvista/pyvista/blob/42d17b1997645f28810a73453716ad08ab155090/pyvista/plotting/plotting.py#L5549-L5570

Things that prevent hanging:
1. removing `plotter.close()`: getting the old plotter destroyed by reference counting doesn't trigger the bug
2. replacing `plotter.close()` with `plotter.mwriter.close()` doesn't trigger the bug (so it's not this exactly causing the hang)
3. opening the GIF plotter with `off_screen=True`.

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Mon Aug 08 16:50:11 2022 CEST

                OS : Linux
            CPU(s) : 8
           Machine : x86_64
      Architecture : 64bit
               RAM : 15.5 GB
       Environment : Python
        GPU Vendor : Intel
      GPU Renderer : Mesa Intel(R) UHD Graphics 620 (WHL GT2)
       GPU Version : 4.6 (Core Profile) Mesa 20.3.5

  Python 3.9.2 (default, Feb 28 2021, 17:03:44)  [GCC 10.2.1 20210110]

           pyvista : 0.37.dev0
               vtk : 9.1.0
             numpy : 1.23.0
           imageio : 2.20.0
           appdirs : 1.4.4
            scooby : 0.5.6
        matplotlib : 3.5.2
           IPython : 8.4.0
          colorcet : 3.0.0
           cmocean : 2.0
        ipyvtklink : 0.2.2
             scipy : 1.8.1
        itkwidgets : 0.32.1
              tqdm : 4.64.0
            meshio : 5.3.4
        jupyterlab : 3.4.4
         pythreejs : Version unknown
--------------------------------------------------------------------------------
```

**Comment by banesullivan** [comment]
Saw this while working on #3889 and tested this on `main`: works fine for me locally. @adeak, can you confirm this is still not working for you?

**Comment by adeak** [comment]
Yup, this was fixed by #3245. Thanks @banesullivan.
